pkg/operator/ceph/cluster/version.go

```diff
@@ -130,8 +130,13 @@ func (c *cluster) detectCephVersion(rookImage, cephImage string, timeout time.Du
 	job := versionReporter.Job()
 	job.Spec.Template.Spec.ServiceAccountName = "rook-ceph-cmd-reporter"
 
-	// Apply the same node selector and tolerations for the ceph version detection as the mon daemons
-	cephv1.GetMonPlacement(c.Spec.Placement).ApplyToPodSpec(&job.Spec.Template.Spec)
+	// Apply the same placement for the ceph version detection as the mon daemons except for PodAntiAffinity
+	placement := c.Spec.Placement.DeepCopy()
+	if mon, ok := placement[cephv1.KeyMon]; ok {
+		mon.PodAntiAffinity = nil
+		placement[cephv1.KeyMon] = mon
+	}
+	cephv1.GetMonPlacement(placement).ApplyToPodSpec(&job.Spec.Template.Spec)
```

Comment:
What you have looks like it works ok, but what about something like this to keep it even simpler?
```
cephv1.GetMonPlacement(placement).ApplyToPodSpec(&job.Spec.Template.Spec)
job.Spec.Template.Spec.Affinity.PodAntiAffinity = nil
```

Reply:
@travisn 
I also think it's simpler but If do so, the `all` placement is ignored for `detect-version`.
I thought that the `all` placement does not need to be ignored here.
Which behavior do you think better?

Reply:
I don't know why anti-affinity would be needed at all for the detect version job. It's a very quick job that would not conflict with any other pod. So I would rather keep it simple unless you see a common scenario that I'm missing.